app/sdil/controllers/variation/PackageOwnController.scala

```diff
@@ -16,33 +16,39 @@
 
 package sdil.controllers.variation
 
-import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.i18n.MessagesApi
 import play.api.mvc.{Action, AnyContent}
 import sdil.actions.VariationAction
 import sdil.config.AppConfig
-import sdil.connectors.SoftDrinksIndustryLevyConnector
 import sdil.controllers.RadioFormController
 import uk.gov.hmrc.http.cache.client.SessionCache
-import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import views.html.softdrinksindustrylevy.register.radio_button
 
-import scala.concurrent.Future
-
 class PackageOwnController(val messagesApi: MessagesApi,
-                           cache: SessionCache,
+                           val cache: SessionCache,
                            variationAction: VariationAction)
                           (implicit config: AppConfig)
-  extends FrontendController with I18nSupport {
-
-  def show: Action[AnyContent] = variationAction { implicit request =>
-    val filledForm = request.data.packageOwn.fold(RadioFormController.form)(RadioFormController.form.fill)
+  extends Journey {
 
-    Ok(radio_button(filledForm, "packageOwnUk", backLink, submitAction))
+  def show: Action[AnyContent] = variationAction.async { implicit request =>
+    val filledForm = RadioFormController.form.fill(
+      request.data match {
+        case d if !d.producer.isProducer && !d.copackForOthers => false
```

Comment:
I think it'd be clearer if this was extracted into a boolean val, e.g.

`val packages = d.producer.isProducer || d.copacksForOthers`